providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java

```diff
@@ -0,0 +1,199 @@
+     * Resolve a string flag
+     */
+    public ProviderEvaluation<String> stringEvaluation(String key, String defaultValue,
+                                                       EvaluationContext ctx) {
+        return resolveGeneric(String.class, key, defaultValue, ctx);
+    }
+
+    /**
+     * Resolve a double flag
+     */
+    public ProviderEvaluation<Double> doubleEvaluation(String key, Double defaultValue,
+                                                       EvaluationContext ctx) {
+        return resolveGeneric(Double.class, key, defaultValue, ctx);
+    }
+
+    /**
+     * Resolve an integer flag
+     */
+    public ProviderEvaluation<Integer> integerEvaluation(String key, Integer defaultValue,
+                                                         EvaluationContext ctx) {
+        return resolveGeneric(Integer.class, key, defaultValue, ctx);
+    }
+
+    /**
+     * Resolve an object flag
+     */
+    public ProviderEvaluation<Value> objectEvaluation(String key, Value defaultValue, EvaluationContext ctx) {
+        return resolveGeneric(Value.class, key, defaultValue, ctx);
+    }
+
+    private <T> ProviderEvaluation<T> resolveGeneric(Class<T> type, String key, T defaultValue,
+                                                     EvaluationContext ctx) {
+        final FeatureFlag flag = flagStore.getFLag(key);
+
+        // missing flag
+        if (flag == null) {
+            return ProviderEvaluation.<T>builder()
+                    .value(defaultValue)
+                    .reason(Reason.ERROR.toString())
+                    .errorCode(ErrorCode.FLAG_NOT_FOUND)
+                    .errorMessage(String.format("requested flag could not be found: %s", key))
+                    .build();
+        }
+
+        // state check
+        if ("DISABLED".equals(flag.getState())) {
+            return ProviderEvaluation.<T>builder()
+                    .value(defaultValue)
+                    .reason(Reason.DISABLED.toString())
+                    .errorCode(ErrorCode.GENERAL)
+                    .errorMessage(String.format("requested flag is disabled: %s", key))
+                    .build();
+        }
+
+
+        final Object resolvedVariant;
+        final String reason;
+
+        if ("{}".equals(flag.getTargeting())) {
+            resolvedVariant = flag.getDefaultVariant();
+            reason = Reason.STATIC.toString();
+        } else {
+            try {
+                final Object jsonResolved = jsonLogicHandler.apply(flag.getTargeting(), ctx.asObjectMap());
+                if (jsonResolved == null) {
+                    resolvedVariant = flag.getDefaultVariant();
+                    reason = Reason.DEFAULT.toString();
+                } else {
+                    resolvedVariant = jsonResolved;
+                    reason = Reason.TARGETING_MATCH.toString();
+                }
+            } catch (JsonLogicException e) {
+                log.log(Level.FINE, "Error evaluating targeting rule", e);
```

Comment:
I think we should be more verbose telling the users that something is off. Maybe this could be a possible configuration option (no need to tackle as part of this PR)
```suggestion
                log.log(Level.WARNING, "Error evaluating targeting rule", e);
```

Reply:
I will update this to warning

Reply:
on second thought, incorrect flag configurations could lead to log overflow. hence I will keep this as `Level.FINE` (which is basically Debug level logs)

Reply:
Good point! We can iterate over with a throttle logger in the future if we see the needs :)

Reply:
Do you think DEBUG is better? I honestly never think of `fine` when I think of log levels.

Reply:
@toddbaert there's no DEBUG type exposed. FINE is the closest level [^1]  we have.

[^1]: https://docs.oracle.com/javase/8/docs/api/java/util/logging/class-use/Level.html

Reply:
That's FINE then. :stuck_out_tongue: